test/connectors/sharedmodel/CalculatorResultSpec.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.sharedmodel
+
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+
+class CalculatorResultSpec extends AnyFreeSpec with Matchers {
+
+  private val flatRate = FlatRate(1, 11, 111, 1111, 1111)
+
+  "FlatRate" - {
+    "marginalRelief" - {
+      "should return 0" in {
+        flatRate.marginalRelief shouldBe 0
+      }
+      "should return corporation tax" in {
+        flatRate.corporationTaxBeforeMR shouldBe 11
```

Comment:
(Optional) I would change this line to use the flatRate value instead:
```scala
flatRate.corporationTaxBeforeMR shouldBe flatRate.corporationTax
```
So we would not need to modify the test if we need to change the flatRate values